### DISCUSSION TITLE
Early Impressions

### DISCUSSION THREAD

**Discussion by BujuHackTosh** [SkyrilHD/Dell-E7250-Hackintosh#42] [General]
I just updated to Monterey and have been using it for about 12 hours, and I thought to share my experience so far of both the OS and EFI.

_The Operating System_
First, there is barely any noticeable difference if you are coming from Big Sur, mostly because the 2015 MacBook Pro t his build is based on does not support most of the Monterey features.

Second, the user experience is somewhat normal and still runs very smoothly. Really impressed. 

The EFI
If you have the Touchscreen model, there is a conflict with the VooDooI2C.Kext /Contents/Plugins/VooDooInput.kext. I have attached a screenshot of the OC Snapshot warning. Should you enable the VOODOO Touch Input kexts, the Hackintosh fails to boot until the aforementioned kext is disabled. 
![Screen Shot 2021-10-26 at 02 37 19](https://user-images.githubusercontent.com/89150049/138999257-3620d713-785b-4a67-8258-f233b07db3cd.png)

There is a noticeable improvement with the Touchpad. Like amazing improvement. I am however rolling back to the previous Touchpad configuration because of two reasons:
1. The touchscreen problem (I am used to multitasking gestures and I am not willing to trade it for the improved trackpad yet. I might use it in a future EFI build.)
2. I have noticed unintentional zooming especially on Browsers, that can be annoying at times. This may be an issue with my trackpad, but I did not have it in my previous EFI so I doubt it. 

Finally, there is a problem with ProperTree not working on Monterey, at least some users including me, are facing this issue. If you encounter this problem, you may try to use [https://manjaro.site/propertree-shows-black-screen-on-macos-monterey-how-to-fix/](url) this fix that worked for me. I used Python 3.9.7 as I noticed that it is choosy with the version of Python.  

Huge thanks and shoutout to SkyrilHD for working tirelessly to improve our beloved pc.

**Comment by BujuHackTosh** [comment]
Update, reverting to the previous Touchpad kexts and enabling touchscreen Kexts did not work. In other words, the touchscreen is definitely broken in the latest OC. I may have to post it in issues?

**Reply by SkyrilHD** [reply]
Hmm very weird. Open an issue and attach an ioreg.

**Comment by SkyrilHD** [comment]
Thank you very much for your feedback.

There is a conflict with two same VoodooInput kexts, so that's why it fails to boot. One should use the VoodooInput that comes with VoodooI2C and disable VoodooInput that's included with VoodooPS2Controller.

However for the next version of the EFI, I will rollback to the old VoodooPS2Controller until I fixed some issues as you said the driver is has a little scrolling issues.

Python in general is broken on Monterey. Installing the Python package fixes the black screen issue but (at least) I cannot open or save any files. However I found a workaround by opening the config file twice as the first one is outputing an error and the second time it actually opens the file. Hopefully a fix is coming soon for Python.

TL;DR
- disable VoodooPS2's VoodooInput and enable VoodooI2C's VoodooInput to fix the touchscreen
- next version will have the old version of the VoodooPS2, so VoodooInput will not become an issue anymore
- as you said, python is broken and an update is needed

**Comment by SkyrilHD** [comment]
So, I finally found the solution for the annoying unintentional zooming and the kext is here: https://github.com/SkyrilHD/VoodooPS2-ALPS/actions/runs/1401508945 Just replace it with the old VoodooPS2.

Btw, did you manage to get your touchscreen working?
